x/exomint/keeper/impl_epochs_hooks.go

```diff
@@ -0,0 +1,78 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	epochstypes "github.com/ExocoreNetwork/exocore/x/epochs/types"
+	types "github.com/ExocoreNetwork/exocore/x/exomint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// EpochsHooksWrapper is the wrapper structure that implements the epochs hooks for the
+// keeper.
+type EpochsHooksWrapper struct {
+	keeper *Keeper
+}
+
+// Interface guard
+var _ epochstypes.EpochHooks = EpochsHooksWrapper{}
+
+// EpochsHooks returns the epochs hooks wrapper.
+func (k *Keeper) EpochsHooks() EpochsHooksWrapper {
+	return EpochsHooksWrapper{k}
+}
+
+// AfterEpochEnd is called after an epoch ends. It is called during the BeginBlock function.
+func (wrapper EpochsHooksWrapper) AfterEpochEnd(
+	ctx sdk.Context, identifier string, number int64,
+) {
+	params := wrapper.keeper.GetParams(ctx)
+	if strings.Compare(identifier, params.EpochIdentifier) == 0 {
+		mintedCoin := sdk.NewCoin(
+			params.MintDenom,
+			params.EpochReward,
+		)
+		mintedCoins := sdk.NewCoins(mintedCoin)
+
+		logger := wrapper.keeper.Logger(ctx)
+		err := wrapper.keeper.MintCoins(ctx, mintedCoins)
+		if err != nil {
+			logger.Error(
+				"AfterEpochEnd",
+				"could not mint coins", err,
+			)
+			return
+		}
+
+		err = wrapper.keeper.AddCollectedFees(ctx, mintedCoins)
+		if err != nil {
+			logger.Error(
+				"AfterEpochEnd",
+				"could not transfer coins", err,
+			)
+			return
+		}
```

Comment:
If the suggestion is to replace `Could not mint coins: $err` with `Failed to mint coins due to $err` it is purely semantics and not necessary because the spirit of the two lines is the same.